lib/dark_worlds_server/runner_supervisor/runner.ex

```diff
@@ -102,14 +102,18 @@ defmodule DarkWorldsServer.RunnerSupervisor.Runner do
 
   @impl true
   def handle_call({:join, user_id, character_name}, _from, state) do
-    {game_state, player_id} = GameBackend.add_player(state.game_state, character_name)
+    case GameBackend.add_player(state.game_state, character_name) do
+      {:ok, {game_state, player_id}} ->
+        state =
+          Map.put(state, :game_state, game_state)
+          |> put_in([:user_to_player, user_id], player_id)
 
-    state =
-      Map.put(state, :game_state, game_state)
-      |> put_in([:user_to_player, user_id], player_id)
+        NewRelic.increment_custom_metric("GameBackend/TotalPlayers", 1)
+        {:reply, {:ok, player_id}, state}
 
-    NewRelic.increment_custom_metric("GameBackend/TotalPlayers", 1)
-    {:reply, {:ok, player_id}, state}
+      {:error, reason} ->
+        {:reply, {:error, reason}, state}
```

Comment:
What happens client side with this change?

Reply:
Mmmm, if the character doesn't exists, the client loses the WebSocket connection and freezes on the "Prepare for Battle" screen 🤔 .